File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string res = "";
    void expand(string& s, int left, int right) {
        while (left >= 0 && right < s.length() && s[left] == s[right]) {
            int val = right - left + 1;
            if (val > res.length()) {
                res = s.substr(left, val);
            }
            left--;
            right++;
        }
    }
    string longestPalindrome(string s) { 
        for(int i=0;i<s.length();i++){
            expand(s,i,i);
            expand(s,i,i + 1);
        }
        
        
        return res; }
};

File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def __init__(self):
        self.res = ""
    
    def expand(self, s:str, left:int, right:int):
        while left >= 0 and right < len(s) and  s[left] == s[right]:
            if right - left + 1 > len(self.res):
                self.res = s[left:right+1]
            left -= 1
            right += 1
    
    def longestPalindrome(self, s: str) -> str:
        for i in range(len(s)):
            self.expand(s, i ,i)
            self.expand(s, i , i+1)
        return self.res
        